core/commands/tags.py

```diff
@@ -0,0 +1,318 @@
+            view.run_command("gs_tags_refresh")
+
+
+class GsTagDeleteCommand(TextCommand, GitCommand):
+
+    """
+    Delete tag(s) in selection.
+    """
+
+    def run(self, edit):
+        # Valid sections are in the Local section
+        valid_ranges = view_section_ranges[self.view.id()][:1]
+
+        lines = util.view.get_lines_from_regions(
+            self.view,
+            self.view.sel(),
+            valid_ranges=valid_ranges
+            )
+
+        items = tuple(line[4:].strip().split() for line in lines if line)
+        if items:
+            for item in items:
+                self.git("tag", "-d", item[1])
+
+            util.view.refresh_gitsavvy(self.view)
+            sublime.status_message(TAG_DELETE_MESSAGE)
+
+
+class GsTagCreateCommand(WindowCommand, GitCommand):
+
+    """
+    Through a series of panels, allow the user to add a tag and message.
+    """
+
+    def run(self):
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        """
+        Prompt the user for a tag name.
+        """
+        self.window.show_input_panel(
+            TAG_CREATE_PROMPT,
+            "",
+            self.on_entered_tag,
+            None,
+            None
+            )
+
+    def on_entered_tag(self, tag_name):
+        """
+        After the user has entered a tag name, prompt the user for a
+        tag message. If the message is empty, use the pre-defined one.
+        """
+        # If the user pressed `esc` or otherwise cancelled.
+        if not tag_name:
+            return
+
+        # TODO: do some validation
+
+        self.tag_name = tag_name
+        self.window.show_input_panel(
+            TAG_CREATE_MESSAGE_PROMPT,
+            "",
+            self.on_entered_message,
+            None,
+            None
+            )
+
+    def on_entered_message(self, message):
+        """
+        Perform `git tag tag_name -F -`
+        """
+        # If the user pressed `esc` or otherwise cancelled
+        if message == -1:
+            return
+
+        if not message:
+            default_message = sublime.load_settings("GitSavvy.sublime-settings").get("default_tag_message")
+            message = default_message.format(tag_name=self.tag_name)
+
+        self.git("tag", self.tag_name, "-F", "-", stdin=message)
+
+
+class GsTagPushCommand(TextCommand, GitCommand):
+
+    """
+    Displays a panel of all remotes defined for the repository, then push
+    selected or all tag(s) to the selected remote.
+    """
+
+    def run(self, edit, push_all=False):
+        if not push_all:
+            # Valid sections are in the Local section
+            valid_ranges = view_section_ranges[self.view.id()][:1]
+
+            lines = util.view.get_lines_from_regions(
+                self.view,
+                self.view.sel(),
+                valid_ranges=valid_ranges
+                )
+
+            self.items = tuple(line[4:].strip().split() for line in lines if line)
```

Comment:
`"string".split()` will output `["string"]`.  Is that the desired behavior here?